### PR TITLE
render list of items in List component

### LINKED ISSUE #1: 1. As a user, I want to read the names of all of the items in my shopping list
## Summary

The most important feature of our app is the ability to look at what's on the list, so the user knows what they need to buy. The `List` component is the best place in our code to do that. In fact, this feature is already partially implemented. Refer to the acceptance criteria in this issue, and the existing code, to understand the work that remains.

The `List` component is rendered on the `/list` page. Because the navigation links are currently broken, you'll have to type that path into your browser's address bar.

## Acceptance criteria
- [x] The `streamListItems` function is used in `App.jsx` to get the items in the current list from the Firestore database
- [x] The list is passed into the `List` component as a prop named `data`
- [ ] In `List.jsx`, the `ListItem` component is used to render the name of each item

## Notes
- ⚠️ Your **only goal** is to make sure that the names of the items render as expected. You might be tempted to modify `ListItem.jsx` to show more information – you will do that **later**. If you want a peek at the future, [this wireframe](https://github.com/the-collab-lab/smart-shopping-list/blob/main/_resources/wireframes/list-populated.png) is what your list could look like closer to the end of the project.
- The data you need is already being passed into the `List` component. What steps do you need to take in order to render it?

### PR DESCRIPTION
**Pull request by 01001101CK**
_For an example of how to fill this template out, [see this Pull Request](https://github.com/the-collab-lab/tcl-3-smart-shopping-list/pull/44)._

## Description
The code we wrote allows the list of items to be rendered in the UI. We use .map to transverse the array. We also added key to give a unique identity to each element. 

## Related Issue
closes #1 

## Acceptance Criteria
- The streamListItems function is used in App.jsx to get the items in the current list from the Firestore database
- The list is passed into the List component as a prop named data
- In List.jsx, the ListItem component is used to render the name of each item

## Type of Changes

|     | Type                       |
| --- | -------------------------- |
|     | :bug: Bug fix              |
| ✓   | :sparkles: New feature     |
|     | :hammer: Refactoring       |
|     | :100: Add tests            |
|     | :link: Update dependencies |
|     | :scroll: Docs              |

### After
Screenshot 2023-04-03 at 4 32 05 PM

## Testing Steps / QA Criteria
Navigate to /list, and a list of items is rendered in the UI.

### CODE REVIEW

**Review by itsoliviasparks — APPROVED** [review]
*chef's kiss*

**Review by itsoliviasparks — APPROVED** [review]

File: src/views/List.jsx [hunk 1]
```diff
@@ -8,10 +8,13 @@ export function List({ data }) {
 			</p>
 			<ul>
 				{/**
-				 * TODO: write some JavaScript that renders the `data` array
+				 * TODO: write some JavaScript that renders the `data` array of objects
 				 * using the `ListItem` component that's imported at the top
 				 * of this file.
 				 */}
```
Comment (vividavila98) [inline src/views/List.jsx]: Go ahead and remove the TODO :)

File: src/views/List.jsx [hunk 2]
```diff
@@ -8,10 +8,13 @@ export function List({ data }) {
...
+				{data.map((data, i) => (
+					<ListItem key={i} name={data.name} />
+				))}
```
Comment (vividavila98) [inline src/views/List.jsx]: Looks great!!

**Review by vividavila98 — APPROVED** [review]
Great job you two! :) Just for practice, it'd be good to fill out the "How to test" section even if it's as simple as "navigate to /list"

**Pull request merged**